### PR TITLE
feat: batch conversion with parallel processing (040)

### PR DESCRIPTION
**Pull request by brianluby**
## Summary

- Adds multi-file batch conversion to `forge convert` (accept 2+ input files, process in parallel via rayon, aggregate status on stderr)
- Preserves single-file backward compatibility — 1 input file routes to the existing `execute()` path unchanged
- Includes output filename collision avoidance (`_2`, `_3` suffixes), `--jobs` flag (0=auto, 1+=explicit thread count), auto-creation of output directories, and panic isolation per file via `catch_unwind`
- Code quality: `FileOutcome` enum eliminates impossible states, `OutputFormat::as_extension()` canonicalizes format mapping, O(1) collision resolution via HashMap

## Test plan

- [x] `cargo test` — 1132 tests pass (0 failures)
- [x] `cargo clippy -- -D warnings` — clean
- [x] Integration tests cover: batch of 3 valid files, single-file compat, filename collisions, `--output` as file error, zero inputs, auto-created output dir, no `--output` (current dir), mixed success/failure, all-failure, status on stderr, `--jobs 1` sequential, error isolation under parallelism, 5-file glob simulation, 100+ file warning
- [ ] Manual: `forge convert policies/*.md --strategy catalog --format json --output out/`

## Follow-up issues

- #46 — Consolidate duplicate empty-input check
- #50 — Remove TOCTOU pre-validation
- #51 — Introduce `ConvertOptions` struct

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Batch conversion: process multiple inputs in one command; single-file behavior kept compatible.
  * Added --jobs flag to control parallelism (0 = auto, >0 = fixed).
  * Automatic output directory creation and filename-collision suffixing.

* **User Experience**
  * Printed batch summary to stderr with per-file status, success/failure counts, and total duration.
  * Returns aggregate non-zero exit when any file fails; per-file failures still produce valid outputs.

* **Documentation**
  * Updated usage notes and examples for batch mode.

* **Chores**
  * Added ignore rule for Agent Brain artifacts.

### CODE REVIEW

**Comment by brianluby** [comment]
> `@brianluby` Sure! Kicking off the review now.
> 
> 🧠 Learnings used
> ✅ Actions performed

@coderabbitai Did this not find anything?

File: specs/040-batch-conversion/contracts/batch.rs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+// --- Data Structures (batch/summary.rs) ---
+
+/// Result of converting a single file in a batch.
+#[derive(Debug)]
+pub struct FileResult {
+    pub input_path: PathBuf,
+    pub output_path: Option<PathBuf>,
+    pub success: bool,
+    pub error_message: Option<String>,
```
[the remark this answers is not in the document]
Comment (brianluby) [reply specs/040-batch-conversion/contracts/batch.rs]: Fixed in 5b583af. Contract doc now uses the `FileOutcome` enum (`Success { output_path }` / `Failure { error_message }`) and `FileResult` with `outcome: FileOutcome`, matching the implementation.

File: src/batch/output_naming.rs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+        let output_path = if claimed.contains(&candidate) {
+            let n = next_suffix.entry(stem.clone()).or_insert(2);
+            let suffixed = base_dir.join(format!("{stem}_{n}.{ext}"));
+            *n += 1;
+            suffixed
+        } else {
+            candidate
+        };
+
```
[the remark this answers is not in the document]
Comment (brianluby) [reply src/batch/output_naming.rs]: Good catch — this was a real bug. Fixed in 5b583af. The suffixed path now re-checks against the `claimed` set in a loop, so `policy_2.md` (natural stem) won't collide with a generated `policy_2.json` suffix. Added a test case for this edge case (`collision_suffix_skips_already_claimed_stem`).
